java-client/build.gradle.kts

```diff
@@ -128,8 +156,8 @@ dependencies {
     val jacksonDatabindVersion = "2.12.6.1"
 
     // Apache 2.0
-    implementation("org.opensearch.client", "opensearch-rest-client", opensearchVersion)
-    testImplementation("org.opensearch.test", "framework", opensearchVersion)
+    implementation("org.opensearch.client", "opensearch-rest-client", "1.3.2")
+    integrationTestImplementation("org.opensearch.test", "framework", opensearchVersion)
```

Comment:
If anything, we should not take this path: `opensearch-rest-client` of one version, `framework` of another version.

Reply:
@reta , I mentioned the 1.3.2 problem in the PR description. And I fully agree that this is not the right approach - it is temporary to have a working build. I aim to be able to remove with https://github.com/opensearch-project/OpenSearch/pull/3181. But, without that merged in, this would not work.

Reply:
Ok, thanks @dlvenable , I still don't understand how it  is supposed to work with `framework` (the `opensearch-rest-client` will be fine): the JDK-8 toolchain will fail trying to use JDK-11 bytecode.

Reply:
This PR builds and runs unit tests against JDK 8. It runs the integration tests using JDK 11 to support the `framework` library.

Reply:
Uh ... thanks @dlvenable , @dblock supporting this JDK-8/JDK-11 "split brain" across project would be quite tricky, we need a really compelling reason to go with this change (https://github.com/opensearch-project/opensearch-java/issues/156 has no evidence it is really needed, @dlvenable please correct me if I am wrong).

Reply:
Same. I really think we're on the right track with things like https://github.com/opensearch-project/opensearch-sdk ... I don't know. Who else do we ask?

Reply:
From the introduction to opensearch-sdk, it is about an SDK for plugin support.

Modern architectures consist of many disparate systems. The clients communicating with OpenSearch are not always the same teams as those running OpenSearch. Or multiple microservices may share an OpenSearch cluster. My primary goal here is to decouple Java clients from OpenSearch server-side decisions. Right now, the Python and .NET clients are already free from the server-side constraints. I think we should do the same for Java.

Currently, a server-side decision to move to JDK 11 (a good choice here) is forcing all clients to use Java 8. I'm unsure of the impact on OpenSearch users, but [according to this post](https://www.infoq.com/news/2021/07/snyk-jvm-2021/), three different surveys show that over half of teams have Java 8 running in production.

Based on some of this feedback, it seems the most appropriate solution is to fully decouple the opensearch-java client from the server-side components. This means removing the low-level-rest-client and the test framework. What thoughts do you have on that approach?

Reply:
> This means removing the low-level-rest-client and the test framework. What thoughts do you have on that approach?

I think the test framework should be gone. The `low-level-rest-client` should probably stay (it is used both by RHLC and `opensearch-java`) but we could have source/target level relaxed to 1.8 (as you did before) , out of all it would be the best option (taking for granted 1.8 as hard requirement).

Reply:
Thanks for your feedback on that @reta . I'll take a look into replacing the test framework in the opensearch-java project.

Reply:
What if we _moved_ `low-level-rest-client` into `opensearch-java`?

Reply:
@dblock we will have dependency from `core` to `opensearch-java`, HLRC uses it [1] (at least)

[1] https://github.com/opensearch-project/OpenSearch/blob/main/client/rest-high-level/build.gradle#L52

Reply:
@dlvenable I think we should put some thoughts into that, in general the server should not depend on its clients, since the server is the one defining the protocols / contracts / etc (and there is only one server vs many clients).

Reply:
this is such a mess; maybe we just take low-level-rest-client out in its own repo? but I dislike that anyway given that the "right" thing to do would be to generate both server and client from _spec_ and not from each-other; I think we're wasting our time

Reply:
Generating the Java client from a spec would be great too. Is there an existing spec for the server from which we could generate this?

Reply:
That's https://github.com/opensearch-project/opensearch-clients/issues/19 !

Reply:
The low-level REST client provides very little value to this library.  I suggest you just git rid of it and use its underlying HTTP transport instead.  One less dependency to worry about.

Reply:
I agree. Maybe you want to take a stab at that @mtimmerm? :)

Reply:
Maybe. I'll look into it